File: Back-end/ai-server/src/main/java/com/lht/pojo/Exercise.java
package com.lht.pojo;

import com.pgvector.PGvector;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Generated;
import org.hibernate.annotations.GenerationTime;
import org.hibernate.annotations.Type;

import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "exercises")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Exercise {

    @Id
    @Column(name = "uuid", updatable = false, nullable = false)
    @Generated(GenerationTime.INSERT)
    private UUID uuid;

    private String name;

    private String force;

    private String difficulty;

    private String mechanic;

    private String equipment;

    @Column(name = "primary_muscles")
    private List<String> primaryMuscles;

    @Column(name = "secondary_muscles")
    private List<String> secondaryMuscles;

    @Column(columnDefinition = "text[]")
    private List<String> instructions;

    @Column(columnDefinition = "text[]")
    private List<String> images;

    private String category;

    @Column(name = "embedding", columnDefinition = "vector(768)")
    @Type(PGvectorType.class)
    private PGvector embedding;
}

File: Front-end/app/lib/models/Exercise.dart
class Exercise {
  final String? uuid;
  final String? name;
  final String? force;
  final String? difficulty;
  final String? mechanic;
  final String? equipment;
  final List<String>? primaryMuscles;
  final List<String>? secondaryMuscles;
  final List<String>? instructions;
  final List<String>? images;
  final String? category;

  Exercise({
    this.uuid,
    this.name,
    this.force,
    this.difficulty,
    this.mechanic,
    this.equipment,
    this.primaryMuscles,
    this.secondaryMuscles,
    this.instructions,
    this.images,
    this.category,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) => Exercise(
    uuid: json['uuid'],
    name: json['name'],
    force: json['force'],
    difficulty: json['difficulty'],
    mechanic: json['mechanic'],
    equipment: json['equipment'],
    primaryMuscles: (json['primaryMuscles'] as List?)?.map((e) => e.toString()).toList(),
    secondaryMuscles: (json['secondaryMuscles'] as List?)?.map((e) => e.toString()).toList(),
    instructions: (json['instructions'] as List?)?.map((e) => e.toString()).toList(),
    images: (json['images'] as List?)?.map((e) => e.toString()).toList(),
    category: json['category'],
  );
}